New WordOfTheDay Spice

tyleryasaka opened the pull request:
![word-of-the-day](https://cloud.githubusercontent.com/assets/6504519/12637326/e97af180-c559-11e5-8b0b-dedd8b4e9da7.png)


IA: https://duck.co/ia/view/word_of_the_day

daxtheduck commented:
@tyleryasaka Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

daxtheduck commented:
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

moollaza commented:
@tyleryasaka Thanks for contributing and welcome to DuckDuckHack!

I noticed you included all the Perl endpoints for the pronunciation, audio, etc but didn't include any JS to use it. Would you like to do that?

tyleryasaka commented:
@moollaza oops, I meant to leave that out of the first iteration. But now that it's been brought up - yeah I think that would be cool. The best way to do this would be to just model the front end for [Dictionary Definition](https://duck.co/ia/view/dictionary_definition), right? I held off on this part to start with because it uses the `base` template group, which (according to the [docs](http://docs.duckduckhack.com/frontend-reference/template-groups.html#base-template-group)) I need permission to use.

tyleryasaka commented:
@moollaza I decided to just hold off on the audio and pronunciation and all that as it is quite involved and I am a bit too busy figure it out now. So I removed that code from the perl file.

Also I changed the template to the base template in order to display the definition similarly to the Dictionary Definition Spice.

How does the code look now?

moollaza commented:
>I decided to just hold off on the audio and pronunciation and all that as it is quite involved and I am a bit too busy figure it out now. So I removed that code from the perl file.

Not a problem, that sounds good! This IA is a good example of how IA's should be able to share code and use each other. Ideally you shouldn't have to do so much replication to get this IA working given that Dictionary already has everything in place!

> Also I changed the template to the base template in order to display the definition similarly to the Dictionary Definition Spice.

Yup that's perfect. This design is already approved by our design team and in-use so you're ok.

> How does the code look now?

Great! I'll get this deployed to beta for more testing

tyleryasaka commented:
ok great :smile:

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=word of the day)
                ![](https://ia-screenshots.s3.amazonaws.com/word_of_the_day_index.png)
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/word_of_the_day_mobile.png)

moollaza commented on lib/DDG/Spice/WordOfTheDay.pm:
```diff
@@ -0,0 +1,28 @@
+
+spice wrap_jsonp_callback => 1;
+
+# for testing, run: DDG_SPICE_WORDNIK_APIKEY=a2a73e7b926c924fad7001ca3111acd55af2ffabf50eb4ae5 duckpan server
```
@tyleryasaka you probably want to remove this :+1:

moollaza commented:
@tyleryasaka Thanks! This LGTM :+1:
